rfs_test/TEST_protocol_details.py

```diff
@@ -4454,104 +4454,131 @@ def Assertion_6_5_31(self, log) :
+    return type_or_context.rsplit('.', 1)[0]
+
+
+def get_annotation_namespace(annotation):
+    """
+    Utility function to extract the namespace from a payload annotation (e.g. @Message.ExtendedInfo)
+    :param annotation: the payload annotation
+    :type annotation: str
+    :return: the namespace
+    """
+    if '@' in annotation:
+        annotation = annotation.rsplit('@', 1)[1]
+    return annotation.rsplit('.', 1)[0]
+
+
+def get_metadata_namespaces(metadata):
+    """
+    Utility function to get all the namespaces from a metadata document and return them as a set()
+    :param metadata: the metadata document
+    :type metadata: schema.Edmx
+    :return: the namespace
+    """
+    metadata_namespaces = set()
+    for reference in metadata.References:
+        for include in reference.Includes:
+            metadata_namespaces.add(include.Namespace)
+    return metadata_namespaces
+
+
+def get_keys_substring(substring, val):
+    """
+    Generator function to recursively find all keys from a dict that containing the substring
+    :param substring: the substring to look for (e.g. '@Message.' to find @Message annotations)
+    :type substring: str
+    :param val: the dict to search
+    :type val: dict
+    :return: all matching keys
+    """
+    if hasattr(val, 'items'):
+        for k, v in val.items():
+            if substring in k:
+                yield k
+            if isinstance(v, dict):
+                for result in get_keys_substring(substring, v):
+                    yield result
+            elif isinstance(v, list):
+                for d in v:
+                    for result in get_keys_substring(substring, d):
+                        yield result
+
+
+###################################################################################################
+# Name: Assertion_6_5_8(self, log)
 # Description:    Referencing Other Schemas
 # The service metadata shall include the namespaces for each of the Redfish resource types, along with
 # the "RedfishExtensions.v1_0_0" namespace.
 ###################################################################################################
-def Assertion_6_5_8(self, log) :
-
+def Assertion_6_5_8(self, log):
     log.AssertionID = '6.5.8'
-    assertion_status =  log.PASS
+    assertion_status = log.PASS
     log.assertion_log('BEGIN_ASSERTION', None)
     authorization = 'on'
 
     relative_uris = self.relative_uris
     rq_headers = self.request_headers()
-    direc = os.getcwd()+'/redfish-1.0.0/json-schema/'
-    ext = '.json'
-    file_dict = {} # Create an empty dict
-    
-    # Select only files with the ext extension
-    txt_files = [i for i in os.listdir(direc) if os.path.splitext(i)[1] == ext]
-    for i in range (0, len(txt_files)):
-        txt_files[i] = txt_files[i].split('.json')[0]
-        #txt_files[i]= txt_files[i][0]
-    # Iterate over your txt files
-    list1 = []
+
+    service_namespaces = set()
+    service_namespaces.add('RedfishExtensions.v1_0_0')
```

Comment:
Need to check that this namespace is aliased to "Redfish" in the metadata document